File: src/main/java/com/braggbnb109/service/SecurityDepositService.java
package com.braggbnb109.service;

import java.util.List;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.domain.Specification;
import org.springframework.http.ResponseEntity;

import com.braggbnb109.domain.SecurityDeposit;
import com.braggbnb109.dto.SecurityDepositDTO;
import com.braggbnb109.dto.SecurityDepositSearchDTO;
import com.braggbnb109.dto.SecurityDepositPageDTO;
import com.braggbnb109.dto.SecurityDepositConvertCriteriaDTO;
import com.braggbnb109.service.GenericService;
import com.braggbnb109.dto.common.RequestDTO;
import com.braggbnb109.dto.common.ResultDTO;
import java.util.List;
import java.util.Optional;





public interface SecurityDepositService extends GenericService<SecurityDeposit, Integer> {

	List<SecurityDeposit> findAll();

	ResultDTO addSecurityDeposit(SecurityDepositDTO securityDepositDTO, RequestDTO requestDTO);

	ResultDTO updateSecurityDeposit(SecurityDepositDTO securityDepositDTO, RequestDTO requestDTO);

    Page<SecurityDeposit> getAllSecurityDeposits(Pageable pageable);

    Page<SecurityDeposit> getAllSecurityDeposits(Specification<SecurityDeposit> spec, Pageable pageable);

	ResponseEntity<SecurityDepositPageDTO> getSecurityDeposits(SecurityDepositSearchDTO securityDepositSearchDTO);
	
	List<SecurityDepositDTO> convertSecurityDepositsToSecurityDepositDTOs(List<SecurityDeposit> securityDeposits, SecurityDepositConvertCriteriaDTO convertCriteria);

	SecurityDepositDTO getSecurityDepositDTOById(Integer securityDepositId);







}

File: braggbnb109_ui/src/services/SecurityDepositService.js
import http from "../http-common"; 

class SecurityDepositService {
  getAllSecurityDeposits(searchDTO) {
    console.log(searchDTO)
    return this.getRequest(`/securityDeposit/securityDeposits`, searchDTO);
  }

  get(securityDepositId) {
    return this.getRequest(`/securityDeposit/${securityDepositId}`, null);
  }

  findByField(matchData) {
    return this.getRequest(`/securityDeposit?field=${matchData}`, null);
  }

  addSecurityDeposit(data) {
    return http.post("/securityDeposit/addSecurityDeposit", data);
  }

  update(data) {
  	return http.post("/securityDeposit/updateSecurityDeposit", data);
  }
  
  uploadImage(data,securityDepositId) {
  	return http.postForm("/securityDeposit/uploadImage/"+securityDepositId, data);
  }




	postRequest(url, data) {
		return http.post(url, data);
      };

	getRequest(url, params) {
        return http.get(url, {
        	params: params
        });
    };

}

export default new SecurityDepositService();
